consensus/types/proto/state_sync.proto

```diff
@@ -30,6 +30,25 @@ message GetBlockResponse {
     consensus.Block block = 2; // The block being provided to the peer
 }
 
+enum StateSyncMessageType {
+    STATE_SYNC_METADATA_REQUEST = 0;
+    STATE_SYNC_METADATA_RESPONSE = 1;
+    STATE_SYNC_GET_BLOCK_REQUEST = 2;
+    STATE_SYNC_GET_BLOCK_RESPONSE = 3;
+
+}
+message StateSyncMessage {
+    StateSyncMessageType msgType = 1;
+
+    oneof message {
```

Comment:
Kind of weird that we have a oneof AND a msg type, but still makes sense. Will keep thinking

Reply:
I am also thinking about it, but it is handy like type of HotStuff Message. I will rethink about it.

Reply:
Thanks for sharing, I updated.